proto/kuksa/val/v2/types.proto

```diff
@@ -74,82 +75,58 @@ enum ErrorCode {
 
 message Metadata {
   // ID field
-  int32 id = 10;  // Unique identifier for the metadata entry
+  int32 id                             = 10;
 
   // Data type
   // The VSS data type of the entry (i.e. the value, min, max etc).
   //
   // NOTE: protobuf doesn't have int8, int16, uint8 or uint16 which means
   // that these values must be serialized as int32 and uint32 respectively.
-  DataType data_type                 = 11;  // [field: FIELD_METADATA_DATA_TYPE]
+  DataType data_type                   = 11;
 
   // Entry type
-  EntryType entry_type               = 12;  // [field: FIELD_METADATA_ENTRY_TYPE]
+  EntryType entry_type                 = 12;
 
   // Description
   // Describes the meaning and content of the entry.
-  optional string description        = 13;  // [field: FIELD_METADATA_DESCRIPTION]
+  string description                   = 13;
 
-  // Comment [optional]
+  // Comment
   // A comment can be used to provide additional informal information
   // on a entry.
-  optional string comment            = 14;  // [field: FIELD_METADATA_COMMENT]
+  string comment                       = 14;
 
-  // Deprecation [optional]
+  // Deprecation
   // Whether this entry is deprecated. Can contain recommendations of what
   // to use instead.
-  optional string deprecation        = 15;  // [field: FIELD_METADATA_DEPRECATION]
+  string deprecation                   = 15;
 
-  // Unit [optional]
+  // Unit
   // The unit of measurement
-  optional string unit               = 16;  // [field: FIELD_METADATA_UNIT]
+  string unit                          = 16;
 
-  // Value restrictions [optional]
-  // Restrict which values are allowed.
-  // Only restrictions matching the DataType {datatype} above are valid.
-  ValueRestriction value_restriction = 17;  // [field: FIELD_METADATA_VALUE_RESTRICTION]
+  // Value restrictions checked/enforced by Databroker
+  ValueRestrictions value_restrictions = 17;
 }
 
-// Value restriction
-//
-// One ValueRestriction{type} for each type, since
-// they don't make sense unless the types match
-//
-message ValueRestriction {
-  oneof type {
-    ValueRestrictionString string        = 21;
-    // For signed VSS integers
-    ValueRestrictionInt signed           = 22;
-    // For unsigned VSS integers
-    ValueRestrictionUint unsigned        = 23;
-    // For floating point VSS values (float and double)
-    ValueRestrictionFloat floating_point = 24;
-  }
+// Restriction fields are checked/enforced by Databroker.
+// They can be either values specified in VSS, or if not present implicit limits for the datatype.
+message ValueRestrictions {
+  Allowed allowed = 1;
+  Value min       = 2;
+  Value max       = 3;
```

Comment:
For me is hard to see as a good alternative `Value allowed_values` if only array types are needed for `allowed` field.
Would it be actually better to have something like this since it makes no sense imo to have allowed and min/max fields at same time?

```proto
message Metadata {
  ..
  // Value restrictions checked/enforced by Databroker
  ValueRestrictions value_restrictions = 1;
}

message ValueRestrictions {
  oneof value {
    Allowed allowed                       = 1;
    MinMaxRestriction min_max_restriction = 2;
  }
}

message MinMaxRestriction {
  Value min = 1;
  Value max = 2;
}
```

Reply:
It's the same as `Value min` and `Value max` being able to contain an array type. That is also not allowed, but enforced by databroker instead of at the proto level.

And it's the same thing everywhere a `Value` is used => it's up to databroker to enforce that the `Value` actually contains what is logically allowed (i.e. by checking the `DataType` of the entry.

Reply:
And the suggested `Allowed` proto message doesn't enforce that the array type matches the datatype of the entry, so databroker has to enforce that manually anyway.

Reply:
I have no strong opinion, but sort of like the simplicity of the suggestion from John:

```
message Metadata {
  ..
  // Value restrictions checked/enforced by Databroker
  Value allowed_values = 17; // Must be of array type
  Value min            = 18;
  Value max            = 19;
}
```

Reply:
Done